macros/src/codegen.rs

```diff
@@ -116,14 +117,36 @@ pub fn generate_code(sm: &ParsedStateMachine) -> proc_macro2::TokenStream {
     // println!("transitions: {:#?}", transitions);
 
     // Map guards, actions and output states into code blocks
-    let guards: Vec<Vec<_>> = transitions
-        .iter()
-        .map(|(_, value)| value.iter().map(|(_, value)| &value.guard).collect())
+    let guards: Vec<Vec<Vec<Option<GuardExpression>>>> = transitions
```

Comment:
Triple vectors here make it _really_ hard to understand what these variables are actually supposed to be. Is there a more logical data structure to use here instead?

Reply:
This is mostly the approach of the original code. 
In the original code the following  expression was used 
```
let guards: Vec<Vec<_>> = transitions ..., 
```
where `_` represented guard data  per (state,event), as the first two layers represent state and event. 
I just made it more explicit by replacing _ with actual type produced by the following query.
We can switch it back to `Vec<Vec<_>>` if this makes readability better, 
but I do not know what else I can do in this case.

Reply:
Eh, rereading it seems fine. Just a maintainability question. I'm not sure what the best structure should be.